---
layout: workshop_main_2day
permalink: /htseq_2018
title: HTseq
header1: Workshop Pages for Students
header2: Informatics on High-Throughput Sequencing Data
image: /site_images/CBW_High-throughput_icon.jpg
keywords: Map sequence data, De Novo assembly tasks, Quantify sequence data
description: Course covers the bioinformatics tools available for managing and interpreting high-throughput sequencing data, where the focus is on Illumina reads although information is applicable to all sequencer reads. 
instructors: Jared Simpson, Mathieu Bourgey, Hamza Farooq
length: 2 days
---
# Welcome <a id="welcome"></a>

Welcome to Informatics on High-Throughput Sequencing Data 2019.  

The course schedule can be found [here](https://bioinformaticsdotca.github.io/htseq_2018_schedule).

Pre-readings, pre-work, and laptop setup instructions can be found [here](https://bioinformaticsdotca.github.io/HTseq_2018_prework).  

We are using Slack for discussion. 

The full course notes are available as a [single PDF]().  

***

# Class Photo
 
Coming soon.

***

# Day 1 <a id="day1"></a>

## Welcome

*<font color="#827e9c">Michelle Brazas</font>*

## Module 1: Introduction to High-throughput Sequencing

*<font color="#827e9c">Jared Simpson</font>* 

[Lecture]()

## Module 2: Data Visualization

*<font color="#827e9c">TBD</font>* 

[Lecture]()

[Lab Practical](https://bioinformaticsdotca.github.io/HTSeq_2018_module2_lab)

## Module 3: Genome Alignment

*<font color="#827e9c">Guillaume Bourque</font>* 

[Lecture]()

[Lab Practical: Connecting to the Cloud](http://bioinformaticsdotca.github.io/AWS_setup)

[Lecture]()  

* We have set up 30 instances on the Amazon cloud - one for each student. In order to log in to your instance, you will need a security certificate. If you plan on using Linux or Mac OS X, please download [this certificate](http://main.oicrcbw.ca/private/CBWNY.pem). Otherwise if you plan on using Windows (with Putty and Winscp), please download [this certificate](http://main.oicrcbw.ca/private/CBWNY.ppk).

[Lab Practical](https://bioinformaticsdotca.github.io/htseq_2018_module3_lab)  

## Integrated Assignment

[Integrated Assignment](https://bioinformaticsdotca.github.io/HTSeq_2018_IA_lab)  

***

# Day 2 <a id="day2"></a>

## Module 4: Small-Variant Calling and Annotation

*<font color="#827e9c">Guillaume Bourque</font>* 

[Lecture]()

[Lab Practical](https://bioinformaticsdotca.github.io/htseq_2018_module4_lab)  

## Module 5: Structural Variant Calling

*<font color="#827e9c">Guillaume Bourque</font>* 

[Lecture]()

[Lab Practical](https://bioinformaticsdotca.github.io/htseq_2018_module5_lab)  

## Module 6: De Novo Assembly

*<font color="#827e9c">Jared Simpson</font>* 

[Lecture]()

[Lab Practical](https://bioinformaticsdotca.github.io/HTSeq_2018_module6_lab)

[Supplemental](https://bioinformaticsdotca.github.io/HTSeq_2018_module6_lab_supplement)

***

Thank you for attending the Informatics on High-Throughput Sequencing Data workshop! Help us make this workshop better by filling out [our survey]().

***

  
  
