---
layout: workshop_main_2day
permalink: /htseq_2019
title: HTseq
header1: Workshop Pages for Students
header2: Informatics on High-Throughput Sequencing Data
image: /site_images/CBW_High-throughput_icon.jpg
keywords: Map sequence data, De Novo assembly tasks, Quantify sequence data
description: Course covers the bioinformatics tools available for managing and interpreting high-throughput sequencing data, where the focus is on Illumina reads although information is applicable to all sequencer reads. 
instructors: Jared Simpson, Guillaume Bourque, Pascale Marquis
length: 2 days
---

# Welcome <a id="welcome"></a>

[Welcome](https://drive.google.com/a/bioinformatics.ca/file/d/11ycZOIktKiIMcSiXRPhgIOnYfMw5GPga/view?usp=sharing) to Informatics for High-throughput Sequencing Data 2019 workshop.  

The course schedule can be found [here](https://bioinformaticsdotca.github.io/htseq_2019_schedule).

Meet your faculty [here](https://drive.google.com/a/bioinformatics.ca/file/d/1vUEx9XPRKPTOrenb_6dfsbsa_j_c5XcS/view?usp=sharing)

**CLASS MATERIALS:** You can download and print the course manual [here](https://drive.google.com/a/bioinformatics.ca/file/d/1blBOuv5Ssh6NumkLdXUEzFGMUom11KgY/view?usp=sharing).  

***

# Pre-Workshop Materials <a id="preworkshop"></a>

## Laptop Setup Instructions

Instructions to setup your laptop can be found [here](https://bioinformaticsdotca.github.io/htseq_2019_installation).

## Pre-workshop Tutorials

Pre-readings and pre-work can be found here [here](https://bioinformaticsdotca.github.io/htseq_2019_prework).  
It is in your best interest to complete these before the workshop.

***

# Class Photo
 
<img src="https://github.com/bioinformaticsdotca/HTseq_2019/blob/master/ClassPhoto_HTseq2019_v2.jpg?raw=true" alt="Class Photo" width="750" />

***

# Day 1 <a id="day1"></a>

## Welcome

*<font color="#827e9c">Michelle Brazas</font>*

## WiFi  

1) Eduroam account; or    
2) Sign up for a McGill 24hr account at : https://www.mcgill.ca/it/get-help/forms/guest-wifi   

## Module 1: Introduction to High-throughput Sequencing

*<font color="#827e9c">Jared Simpson</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1c9GhQEzqpxhpEg7FMjI3uG7H-s9Zf-P3/view?usp=sharing)

## Module 2: Data Visualization

*<font color="#827e9c">Rob Syme</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1Hy8x5ZexnP9Wvri0vGZzHLLRiIrXU8Ia/view?usp=sharing)

[Lab Practical](https://bioinformaticsdotca.github.io/htseq_2019_module2_lab)

[Lab Answer Key](https://bioinformaticsdotca.github.io/htseq_2019_module2_lab_answers)   

## Places to Eat

- Pizza Bros https://www.pizzeriabros.ca (1212 Avenue Union)   
- Falafel avenue http://falafelavenue.com/en/ (2020, rue Robert-Bourassa (at University)   
- Japote: http://www.japote.ca (1000 Rue Sherbrooke Ouest)   
- Opanio: https://www.opianomtl.com (1115 Sherbrooke St W)   
- KoaLua: http://www.koalua.ca (1212 Avenue Union)   

## Module 3: Genome Alignment

*<font color="#827e9c">Guillaume Bourque</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1APBQV9SvvU2PUiNS3o4tEBz9DAeuHNmB/view?usp=sharing)

[Lab Practical: Connecting to the Cloud](http://bioinformaticsdotca.github.io/AWS_setup)

[Lecture - AWS](https://drive.google.com/a/bioinformatics.ca/file/d/18avla7ZXf7j4zBvKYjoYiZnuf7kbPzeP/view?usp=sharing)  

* We have set up instances on the Amazon cloud - one for each student. In order to log in to your instance, you will need a security certificate. The certificates will be uploaded to Slack. If you plan on using Linux or Mac OS X, Please use CBWHT.pem; Otherwise if you plan on using Windows (with Putty), please use CBWHT.ppk.

You can also download from here. [Mac/Linux certificate](http://ht.oicrcbw.ca/private/CBWHT.pem), [Windows certificate](http://ht.oicrcbw.ca/private/CBWHT.ppk).

[Lab Practical](https://bioinformaticsdotca.github.io/htseq_2019_module3_lab)  

## Integrated Assignment

[Integrated Assignment](https://bioinformaticsdotca.github.io/htseq_2019_IA)    

***

# Day 2 <a id="day2"></a>

## Module 4: Small-Variant Calling and Annotation

*<font color="#827e9c">Guillaume Bourque</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/100_N2hRYbybxPx_HX7KFMvU8N7s3OfRd/view?usp=sharing)

[Lab Practical](https://bioinformaticsdotca.github.io/htseq_2019_module4_lab)  

## Module 5: Structural Variant Calling

*<font color="#827e9c">Guillaume Bourque</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1foZ4PoOeID9JCCnIQhgEhsCVcKyIHZu2/view?usp=sharing)

[Lab Practical](https://bioinformaticsdotca.github.io/htseq_2019_module5_lab)  

## Module 6: De Novo Assembly

*<font color="#827e9c">Jared Simpson</font>* 

[Lecture](https://drive.google.com/a/bioinformatics.ca/file/d/1_WWMbe_0s328OJ1ZPEZSs6l3-NyhgBxB/view?usp=sharing)

[Lab Practical](https://bioinformaticsdotca.github.io/htseq_2019_module6_lab)

[Supplemental](https://bioinformaticsdotca.github.io/htseq_2019_module6_lab_supplement)

## Post Workshop 

Post workshop content for data sets, docker containers, AWS image, tools and install instructions can be found [here](https://bioinformaticsdotca.github.io/htseq_2019_post-workshop)  

***

Thank you for attending the Informatics on High-Throughput Sequencing Data workshop! Help us make this workshop better by filling out [our survey]().

***



  
  
