Here are the main solutions:

 * You could look at it manually through whatever text editor but it will be very complicated to make sens of the data
 * You could use specific database (i.e Gemini) to explore and annotate your vcf
 * But one of the best way to make sens of your data is to look at variants in a genomic browser which can also display the read information (i.e. IGV or the UCSC genome browser). 
 * Delly also offer add-on tools to vizualise its calls (i.e. Delly-maze and Delly-suave).


